### DISCUSSION TITLE
Параметры запуска по умолчанию для OpenWRT

### DISCUSSION THREAD

**Discussion by HiHat** [Waujito/youtubeUnblock#201] [Q&A]
Скрипты запуска для OpenWRT устанавливают ряд параметров с которыми работает хуже, чем при запуске без параметров (вручную).
Как узнать какие настройки действуют при запуске без параметров?

**Comment by Waujito** [comment]
Странно, но кстати вероятность что-то упустить имеется. Я сейчас вообще хочу отказаться от предпрописанных дефолтных аргументов для опенврт в пользу простой передачи списка аргументов, как обычному command line анблоку.

**Reply by andrrrw** [reply]
Это правильное решение на мой взгляд. 
Хочется самому ставить флаги, а предустоновленные в конфиге создают некоторую путаницу
Проще будет сделать передачу аргументов в строку
Набор предустановленных аргументов можно оставить где-нибудь отдельным файлом

**Comment by MayanKoyote** [comment]
Смотрю последние коммиты, #202:
Это в смысле что вместо крыжиков будет такой типа как "блокнот" в духе "пиши с ошибками что в голову взбредёт"?
Зачем тогда вообще такой luci-app нужен будет?

**Reply by Waujito** [reply]
luci-app нужен, чтобы можно было использовать анблок без ssh до роутера. Предыдущая версия была конечно интересной идеей, но анблок обновляется, и я стараюсь уменьшить количество того, что надо менять при добавлении нового флага. Раньше дополнительными были модуль ядра, ридми и luci-app. Сейчас - просто ридми. Дальше по списку - добавление секций в анблок. Мне придётся с нуля перекраивать luci-app, чтобы это всё органично выглядело. 

И еще при экспериментах, обновлениях всё равно список флагов передаётся как список флагов, и пользователям важна лишь конечная рабочая версия.

"блокнот" выглядит так:
![image](https://github.com/user-attachments/assets/a8f2324c-6fe2-4aa1-8daf-a9028cdf5378)

**Reply by MayanKoyote** [reply]
Оставлю здесь своё безусловное негодование по этому поводу.
Раньше было лучше.
Причину (а именно:  "уменьшить количество того, что надо менять при добавлении нового") понимаю, но раньше было луше.

Опять же для чего эти новые флаги.
Если исключительно для Дискорда, то что они вообще делают в ПО, называющемся "youtubeUnblock" (а не "ydUnblock", к примеру)? Я понимаю что их туда практически можно. Но теоретически - зачем?
Так же не стоит забывать что если Y нас просто "плохо работает, ну вот так вот", то D уже официально ограничен. Легальная чистота теряется. Не все пользователи одинаковы. "За скобки" бы такое выносить, плагинами, модулями, отдельной версией, как угодно. Серьёзно.

**Reply by Waujito** [reply]
> Причину (а именно:  "уменьшить количество того, что надо менять при добавлении нового") понимаю, но раньше было луше.

В целом, если интерактивные тыкалки так сильно нравились, я могу их вернуть, но придётся потратить достаточно много времени для обновления luci. Секции - не единственный его недостаток, я знаю еще много возможных проблем с ним, а lua интерфейс может очень быстро исчерпать себя, и придётся переходить на js.

> Опять же для чего эти новые флаги.
> Если исключительно для Дискорда

Не только. Различные политики для различных случаев это мощная технология, которая позволит правильно развиваться дальше. 

Но пока я остаюсь любителем ютуба, он остаётся главным направлением :)

**Reply by MayanKoyote** [reply]
А если "совместить"?
И настройка крыжиками и настройка "блокнотом".
Как режимы настройки базовый/продвинутый. Возможно в 2-х табах. С общим подвалом где статус, кнопки и лог (и возможно результат - текущие аргументы в виде полной строки).
В базовом (с крыжиками) проработать тексты к этим крыжикам для большей компактности/ёмкости текстов.

Понимаю что это еще больше работы.
В lua не разбираюсь, не программист. Читать могу, видеть потенциальные проблемы - нет.

**Reply by Waujito** [reply]
Да, я и сам думаю совместить аргументы строкой для copy-paste и интерактивные. Пробую перенести всё на js, он гораздо удобнее, чем deprecated lua + там поддержка риал тайм логов и крутые кнопочки. И там потенциально можно задать очень красивый способ конфигурации секций.

**Reply by Waujito** [reply]
@MayanKoyote как будто бы получилось сделать красивый luci-app. Можете попробовать в этом акшнсе https://github.com/Waujito/youtubeUnblock/actions/runs/12455144585 обновлять и luci-app, и обычный анблок

**Reply by MayanKoyote** [reply]
Не раньше вечера понедельника смогу только. Отпишусь обязательно.

**Reply by HiHat** [reply]
> Можете попробовать в этом акшнсе https://github.com/Waujito/youtubeUnblock/actions/runs/12455144585 обновлять и luci-app, и обычный анблок

Т.к. поменялся формат файла /etc/config/youtubeUnblock, то после обновления со старой версии в luci-app отсутствует Default section.
Если вручную удалить старый конфиг и создать новый командой /usr/share/youtubeUnblock/youtubeUnblock_defaults.sh, то после этого, на первый взгляд, все работает нормально!
Единственное, что заметил в luci-app - невозможность прокрутить журнал syslog в начало, т.к. он постоянно автоскроллируется в конец.

**Reply by Waujito** [reply]
> автоскроллируется в конец.

Надо задокументировать это будет. автоскролл отключается кнопкой refreshing сверху. 

Раз возникают проблемы со старым конфигом, придётся вернуть кнопку резета

**Comment by Ser9ei** [comment]
В OpenWrt версии сервиса, если явно не указывать параметр silent, в лог выдается информация по запрашиваемым доменам, можно инвертировать такое поведение и явно указывать атрибут, для вывода такой информации в лог? Лог быстро заполняется событиями, хотя эта информация скорее для разбора ошибок и расширенного логирования подходит больше. 
Достаточно, наверно, чтоб silent был в параметрах запуска по умолчанию, сейчас при сбросе ‘конфигурации’ сервис стартует без него.
Такое поведение и у kmod сервиса.

Silent атрибут не смог найти в интерфейсе обновленного luci-app модуля, его там нет?

**Reply by Waujito** [reply]
Да флаг там затёрся чуть чуть, я исправил. В целом лог опенврт это просто лог, и есть сервисы, которые тоже туда очень любят писать. Какой-нибудь doh как вариант. Да и для пользователя программа, которая делает лог выглядит куда приятнее, чем просто мёртвое окно. Так что нет, по дефолту сайлента не будет. А флажок в интерфейсе я вернул (где нибудь на последних акшнсах должен быть)
